sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Queues/src/QueueCausalityManager.cs

```diff
@@ -28,10 +29,12 @@ internal class QueueCausalityManager
         private const string ParentGuidFieldName = "$AzureWebJobsParentId";
 
         private readonly ILogger<QueueCausalityManager> _logger;
+        internal readonly JsonSerializerSettings jsonSerializerSettings;
 
-        public QueueCausalityManager(ILoggerFactory loggerFactory)
+        public QueueCausalityManager(ILoggerFactory loggerFactory, JsonSerializerSettings settings = default)
```

Comment:
We should not be exposing Newtonsoft.Json directly from public APIs. Use an abstraction of only settings we care about instead to marshal values between types.

Reply:
Same response to this.

Reply:
We already have precedence for this in the Service Bus extensions. We should probably ask the architects.